pysd/py_backend/functions.py

```diff
@@ -1047,9 +1047,9 @@ def _format_return_timestamps(self, return_timestamps=None):
             # so we have to add an extra period to make sure we get that value in what
             # numpy's `arange` gives us.
             return_timestamps_array = np.arange(
-                self.components.initial_time(),
-                self.components.final_time() + self.components.saveper(),
-                self.components.saveper(), dtype=np.float64
+                self.time(),
+                self.components.final_time() + self.components.saveper()/2,
```

Comment:
what's going on here?

Reply:
If the user changes the initial time using
```
model.run(initial_conditions=(new_initial_time, {}))
```
`model.set_initial_condition` will be called first, this will change the `model.time` value to `new_initial_time`.
Later `model._format_return_timestamps` will be called. However, the model will not start from `initial_time()` but from the `time` value whitch was set to `new_initial_time`.
Therefore, default `return_timestamps` should start in `self.time` and not `self.components.initial_time`.
Changing `self.components.saveper()` by `self.components.saveper()/2` in the arrange will not make any difference in most of the cases, but can prevent having the last timestamp = final_tima + savepar if there is a small numerical error.
Then I change np.float64 by regular float, as the second one has a savest representation of a 64 bits float.

Reply:
Thanks, that makes sense. =)